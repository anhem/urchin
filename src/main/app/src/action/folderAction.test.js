import thunk from 'redux-thunk'
import configureMockStore from 'redux-mock-store'
import fetchMock from 'fetch-mock'
import {
    confirmEncryptedFolder,
    createFolder,
    deleteFolder,
    getFolder,
    getFolders,
    setEncryptedFolderPassphrase,
    setFolder
} from "./folderAction";

describe('folderAction', () => {

    const headers = {'content-type': 'application/json'};
    const getFoldersResponse = {};
    const folderId = 1;
    let store;

    beforeEach(() => {
        store = configureMockStore([thunk])({});
    });

    afterEach(() => {
        fetchMock.reset();
        fetchMock.restore();
    });

    it('getFolders is successful', () => {

        fetchMock.getOnce('/api/folders', getFoldersResponse);

        return store.dispatch(getFolders()).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        })
    });

    it('getFolder is successful', () => {

        fetchMock.getOnce('/api/folders/' + folderId, getFoldersResponse);

        return store.dispatch(getFolder(folderId)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        })
    });

    it('setFolder is successful', () => {

        const folder = {path: 'folderpath'};

        store.dispatch(setFolder(folder));
        expect(store.getActions()).toMatchSnapshot();
    });

    it('createFolder is successful', () => {

        const folder = {path: 'folderpath'};

        fetchMock.postOnce('/api/folders/create', folder);

        return store.dispatch(createFolder(folder)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        })
    });


    it('deleteFolder is successful', () => {

        fetchMock.deleteOnce('/api/folders/' + folderId, {});

        return store.dispatch(deleteFolder(folderId)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        })
    });

    it('setEncryptedFolderPassphrase is successful', () => {

        const passphrase = {passphrase: 'random'};

        store.dispatch(setEncryptedFolderPassphrase(passphrase));
        expect(store.getActions()).toMatchSnapshot();
    });

    it('confirmEncryptedFolder is successful', () => {

        const passphrase = 'random';
        const data = {
            folderId: folderId,
            passphrase: passphrase
        };

        fetchMock.postOnce('/api/folders/mount', data);

        return store.dispatch(confirmEncryptedFolder(folderId, passphrase)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        })
    });

});