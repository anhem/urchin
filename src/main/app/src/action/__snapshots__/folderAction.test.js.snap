// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`folderAction confirmEncryptedFolder is successful 1`] = `
Array [
  Object {
    "type": "CONFIRM_NEW_ENCRYPTED_FOLDER",
  },
  Object {
    "data": Object {
      "folderId": 1,
      "passphrase": "random",
    },
    "type": "CONFIRM_NEW_ENCRYPTED_FOLDER_SUCCESS",
  },
]
`;

exports[`folderAction createFolder is successful 1`] = `
Array [
  Object {
    "type": "SAVE_FOLDER",
  },
  Object {
    "data": Object {
      "path": "folderpath",
    },
    "type": "SAVE_FOLDER_SUCCESS",
  },
]
`;

exports[`folderAction deleteFolder is successful 1`] = `
Array [
  Object {
    "type": "DELETE_FOLDER",
  },
  Object {
    "data": Object {},
    "type": "DELETE_FOLDER_SUCCESS",
  },
]
`;

exports[`folderAction getFolder is successful 1`] = `
Array [
  Object {
    "type": "GET_FOLDER",
  },
  Object {
    "data": Object {},
    "type": "GET_FOLDER_SUCCESS",
  },
]
`;

exports[`folderAction getFolders is successful 1`] = `
Array [
  Object {
    "type": "GET_FOLDERS",
  },
  Object {
    "data": Object {},
    "type": "GET_FOLDERS_SUCCESS",
  },
]
`;

exports[`folderAction setEncryptedFolderPassphrase is successful 1`] = `
Array [
  Object {
    "data": Object {
      "passphrase": "random",
    },
    "type": "SET_CONFIRM_NEW_ENCRYPTED_FOLDER",
  },
]
`;

exports[`folderAction setFolder is successful 1`] = `
Array [
  Object {
    "data": Object {
      "path": "folderpath",
    },
    "type": "SET_FOLDER",
  },
]
`;
